### ISSUE TITLE
Configuration doesn't match VS Code settings editor expectations

### ISSUE THREAD

**Issue by segevfiner** [Azure/vscode-kubernetes-tools#622]
VS Code settings editor expects flat settings with dotted keys, but this extensions uses a single key as an object with the rest of the keys under it (And with a duplicated prefix, you get `vs-kubernetes` twice, once in the object key, and once in the setting key itself). This results in the settings editor not working correctly for this extensions settings.

![screenshot](https://user-images.githubusercontent.com/24731903/65506225-c79e0b00-ded3-11e9-9da6-5bcb94a26b55.png)

P.S. Some extensions took the liberty of using a generic prefix like `vscode-python` simple uses `python.` as a prefix for it's settings rather then `vscode-python`, `vs-python`, etc. Since this is an official extension by Microsoft, you can consider doing so too for consistency with the others. e.g. using `kubernetes.namespace`, etc.

**Comment by itowlson** [comment]
Yes, this frustrates me a bit.  The reasons are historical - when Brendan wrote the very first k8s extension he did it this way I think because he was learning his way, and by the time we realised that the flat way was 'better', there were lots of installations already out there!  So in order to migrate to the flat style, we need to figure out how to handle backward compatibility... not rocket science, but not something we have prioritised yet...
